### DISCUSSION TITLE
Updating fill_between data

### DISCUSSION THREAD

**Discussion by finnstadel** [highfestiva/finplot#149] [Q&A]
Thanks for the great library. I'm putting together an interactive version of a program using your library and PyQt5.
Based on your example here: https://github.com/highfestiva/finplot/blob/master/finplot/example-embed.py#L10

Besides normal plots and volume_ocv I have some line segments (.add_line) and fill_between).
I need to update the window when selecting new data. For .plot and .ocv_volume (and I think  also candlestick_ochl) there are .update functions. But what about lines and fill_between ? Currently I'm using .remove_line - first - for the lines, but I'm not sure what to do with the fill_between. Any advice ?

PS. I'm still learning python, pandas, numpy, your lib etc . Quite a mouthfull coming from a more traditional C, matlab etc background (hardware engineer). Getting to the top of the hill of the learning curve though ;-)

**Comment by highfestiva** [comment]
Good question, hadn't thought of that. Will look in to it in the coming days.

**Reply by finnstadel** [reply]
Sounds good.
In the example, it's using plots and other things as global variables (you need to add "global plots" in updateGUI in order to modify it inside. There should be a way to use all local variables.. hm..
for now I added this to remove the top and bottom lines and the fill between.
```python
if fills:
    ax.removeItem(fills[0])
    ax.removeItem(plots[10])
    ax.removeItem(plots[11])
```

**Reply by highfestiva** [reply]
As long as you use it for its intended purpuse, backtesting, you only need local vars. It's only for real-time you use the global vars. You could create a class and use member variables, if you prefer that.
